core/src/main/java/io/paradoxical/cassieq/admin/resources/api/v1/PermissionsResource.java

```diff
@@ -93,7 +94,8 @@ public Response generateAuthUrl(@NotNull GetAuthQueryParamsRequest request) thro
                         request.getAccountName(),
                         authorizationLevels,
                         request.getStartTime(),
-                        request.getEndTime());
+                        request.getEndTime(),
```

Comment:
lets decouple the data from the action. lets create a signature payload that can be passed to a signer and a query param generator. the query param generaotr can expose like a 

```
var signatureParams = new SigningParameters(
                        request.getAccountName(),
                        authorizationLevels,
                        request.getStartTime(),
                        request.getEndTime(),
                        request.getQueueName());
String queryParams = new QueryParamGenerator(signatureParams, mac)
```

or something